File: src/main/java/syntaxfx/Lexer.java
package syntaxfx;

/**
 * This class represents a Lexer. All lexers used by SyntaxFX must subclass this class.
 */
public abstract class Lexer {
    
    /**
     * Sets the string to be scanned. All lexers must override this method.
     */
    public abstract void setString(String string);

    /**
     * Gets the next token found by the lexer. Returns null when no tokens remain. All lexers must override this method.
     */
    public abstract Token getNextToken();
}

File: src/main/scala/syntaxfx/Lexer.scala
package syntaxfx

/**
  * This class represents a Lexer. All lexers used by SyntaxFX must subclass this class.
  */
abstract class Lexer {

    /**
      * Sets the string to be scanned. All lexers must override this method.
      */
    def setString(string: String): Unit

    /**
      * Gets the next token found by the lexer. Returns null when no tokens remain. All lexers must override this method.
      */
    def getNextToken(): Token

}
